treepoem/__main__.py

```diff
@@ -0,0 +1,35 @@
+# -*- encoding:utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import sys
+from textwrap import fill
+
+from . import generate_barcode
+from .data import barcode_types
+
+supported_barcode_types = (
+    'Supported barcode types are:\n'
+    + fill(', '.join(sorted(barcode_types)), initial_indent='    ', subsequent_indent='    ')
+)
+
+
+def main():
+    p = argparse.ArgumentParser(epilog=supported_barcode_types)
+    p.add_argument('-t', '--type', default='qrcode', help='Barcode type (default (%(default)s))')
+    p.add_argument('-f', '--fmt', help='Output format (default is based on file extension)')
```

Comment:
`--format` would be more standard, if users want to save characters they use `-f`